CMakeLists.txt

```diff
@@ -0,0 +1,323 @@
+#CONFIGURE                            copy.
+#CONFIGURE    --enable-flop-counter   Support (approximate) FLOP counting by the library.
+#CONFIGURE                            CXXGEN must support C++11!
+#CONFIGURE    --enable-profile        Turn on profiling instrumentation of the library.
+#CONFIGURE                            CXXGEN must support C++11!
+#CONFIGURE    --enable-mpfr           Use MPFR library to test Libint integrals in high
+#CONFIGURE                            precision.
+#CONFIGURE  
+#CONFIGURE  Optional Packages:
+#CONFIGURE    --with-api-prefix       Prepend this string to every name in the library API
+#CONFIGURE                            (except for the types).
+#CONFIGURE    --with-vector-length=N  Compute integrals in vectors of length N.
+#CONFIGURE    --with-vector-method    Specifies how to vectorize integrals. Allowed values
+#CONFIGURE                            are 'block' (default), and 'line'.
+#CONFIGURE    --with-build-id         Gives an identifier for the build.
+#CONFIGURE    --with-libint-exportdir Specifies export directory name. Default is
+#CONFIGURE                            libint-<version>.
+#CONFIGURE    --with-pkgconfigdir     Specifies the location to put pkg-config's data
+#CONFIGURE                            file. Default is $(libdir)/pkgconfig.
+#CONFIGURE    --with-cmakedir         Specifies the location to put FindLibint2.cmake
+#CONFIGURE                            file. Default is $(libdir)/cmake/libint2.
+#CONFIGURE    --with-real-type        Compute all integrals using the given real type           LIBINT_USER_DEFINED_REAL
+#CONFIGURE                            (default: double).
+#CONFIGURE    --with-real-type-includes                                                         LIBINT_USER_DEFINED_REAL_INCLUDES
+#CONFIGURE                            Additional includes necessary to use the real type
+#CONFIGURE                            (default: none).
+
+######################## Process & Validate Options ###########################
+include(autocmake_safeguards)
+include(CheckFunctionExists)
+include(int_orderings)
+include(int_am)
+
+check_function_exists(posix_memalign HAVE_POSIX_MEMALIGN)
+if (HAVE_POSIX_MEMALIGN)
+    set(LIBINT2_ALIGN_SIZE "0" CACHE STRING "(EXPERT) if posix_memalign is available, this will specify alignment of Libint data, in units of sizeof(LIBINT2_REALTYPE). Default is to use built-in heuristics")
+endif()
+
+# C++11 required for now. enforced in targets
+set(LIBINT_HAS_CXX11 1)
+set(LIBINT_USE_TYPE_TRAITS 1)
+set(LIBINT_USE_MEMORY 1)
+set(LIBINT_USE_ARRAY 1)
+set(LIBINT_HAS_STD_TYPE_TRAITS 1)
+set(LIBINT_HAS_STD_SHARED_PTR 1)
+set(LIBINT_HAS_STD_MAKE_SHARED 1)
+set(LIBINT_HAS_STD_ARRAY 1)
+set(LIBINT_ARRAY_HAS_FILL 1)
+
+# req'd for now (no FindGMP.cmake, though one at Elemental project)
+find_package(MPFR REQUIRED)
+if (TARGET MPFR::MPFRXX)
+    set(LIBINT_HAS_MPFR 1)
+endif()
+
+find_package(Boost 1.57 REQUIRED)
```

Comment:
I wonder why Boost 1.57+ is necessary (#180).

Reply:
@e-kwsm variadic PP is broken when using clang until boost 1.57: see note here https://github.com/evaleev/libint/blob/master/configure.ac#L1318 and https://www.boost.org/users/history/version_1_57_0.html